gateway/src/main/java/com/hurence/webapiservice/http/ingestion/ImportRequestParser.java

```diff
@@ -0,0 +1,98 @@
+package com.hurence.webapiservice.http.ingestion;
+
+import com.hurence.webapiservice.http.grafana.util.DateRequestParserUtil;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+
+import static com.hurence.historian.modele.HistorianFields.METRIC_NAME_REQUEST_FIELD;
+import static com.hurence.historian.modele.HistorianFields.POINTS_REQUEST_FIELD;
+
+public class ImportRequestParser {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ImportRequestParser.class);
+
+
+    public ResponseAndErrorHolder checkAndBuildValidHistorianImportRequest(JsonArray metricsParam) throws IllegalArgumentException {
+        if (null == metricsParam) {
+            throw new NullPointerException("Null request body");
+        }else if (metricsParam.isEmpty()) {
+            throw new IllegalArgumentException("Empty request body");
+        }
+        ResponseAndErrorHolder responseAndErrorHolder = new ResponseAndErrorHolder();
+        for (Object object : metricsParam) {
+            JsonObject timeserie = (JsonObject) object;
+            if (!(timeserie.containsKey(METRIC_NAME_REQUEST_FIELD))) {
+                throw new IllegalArgumentException("Missing a name for at least one metric");
+            } else if ((timeserie.getValue(METRIC_NAME_REQUEST_FIELD) == null) && (timeserie.getValue(POINTS_REQUEST_FIELD) != null)) {
+                int numPoints = timeserie.getJsonArray(POINTS_REQUEST_FIELD).size();
+                responseAndErrorHolder.errorMessages.add("Ignored "+ numPoints +" points for metric with name 'null' because this is not a valid name");
+                continue;
+            } else if (!(timeserie.getValue(METRIC_NAME_REQUEST_FIELD) instanceof String)) {
+                throw new IllegalArgumentException("A name is not a string for at least one metric");
+            } else if (!(timeserie.containsKey(POINTS_REQUEST_FIELD)) || (!(timeserie.getValue(POINTS_REQUEST_FIELD) instanceof JsonArray)) || (timeserie.getValue(POINTS_REQUEST_FIELD)==null))
```

Comment:
Here you should not theow the same message depending on value of POINTS_REQUEST_FIELD, If it exist but it is not an array you should say it.

Reply:
done